File: Examples/CSharp/AsposePDF/Pages/UpdateDimensions.cs
//////////////////////////////////////////////////////////////////////////
// Copyright 2001-2013 Aspose Pty Ltd. All Rights Reserved.
//
// This file is part of Aspose.Pdf. The source code in this file
// is only intended as a supplement to the documentation, and is provided
// "as is", without warranty of any kind, either expressed or implied.
//////////////////////////////////////////////////////////////////////////
using System.IO;

using Aspose.Pdf;

namespace CSharp.AsposePdf.Pages
{
    public class UpdateDimensions
    {
        public static void Run()
        {
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_AsposePdf_Pages();

            //open document
            Document pdfDocument = new Document(dataDir + "UpdateDimensions.pdf");

            //get page collection
            PageCollection pageCollection = pdfDocument.Pages;

            //get particular page
            Page pdfPage = pageCollection[1];

            // set the page size as A4 (11.7 x 8.3 in) and in Aspose.Pdf, 1 inch = 72 points
            // so A4 dimensions in points will be (842.4, 597.6)
            pdfPage.SetPageSize(597.6, 842.4);

            // save the updated document
            pdfDocument.Save(dataDir + "UpdateDimensions_out.pdf");

        }
    }
}

File: Examples/VisualBasic/AsposePDF/Pages/UpdateDimensions.vb
Imports System.IO
Imports System
Imports Microsoft.VisualBasic
Imports Aspose.Pdf

Namespace AsposePDF.Pages
    Public Class UpdateDimensions
        Public Shared Sub Run()
            ' ExStart:UpdateDimensions
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_AsposePdf_Pages()

            ' Open document
            Dim pdfDocument As New Document(dataDir & Convert.ToString("UpdateDimensions.pdf"))

            ' Get page collection
            Dim pageCollection As PageCollection = pdfDocument.Pages

            ' Get particular page
            Dim pdfPage As Page = pageCollection(1)

            ' Set the page size as A4 (11.7 x 8.3 in) and in Aspose.Pdf, 1 inch = 72 points
            ' So A4 dimensions in points will be (842.4, 597.6)
            pdfPage.SetPageSize(597.6, 842.4)

            dataDir = dataDir & Convert.ToString("UpdateDimensions_out_.pdf")
            ' Save the updated document
            pdfDocument.Save(dataDir)
            ' ExEnd:UpdateDimensions
            System.Console.WriteLine(Convert.ToString(vbLf & "Page dimensions updated successfully." & vbLf & "File saved at ") & dataDir)

        End Sub
    End Class
End Namespace
